parser/schema.go

```diff
@@ -133,39 +143,16 @@ func (p *parser) resolveType(pkg *est.Package, file *est.File, expr ast.Expr) *s
 		p.err(expr.Pos(), "cannot use function types in Encore schema definitions")
 
 	default:
+		if resolvedType := additionalTypeResolver(p, pkg, file, expr); resolvedType != nil {
+			return resolvedType
+		}
+
 		p.errf(expr.Pos(), "%s is not a supported type; got %+v", types.ExprString(expr), reflect.TypeOf(expr))
 	}
 
 	panic(bailout{})
 }
 
-// parseDecl parses the type from a package declaration.
-func (p *parser) parseDecl(pkg *est.Package, d *names.PkgDecl) *schema.Type {
-	key := pkg.ImportPath + "." + d.Name
-	decl, ok := p.declMap[key]
-	if !ok {
-		// We haven't parsed this yet; do so now.
-		// Allocate a decl immediately so that we can properly handle
-		// recursive types by short-circuiting above the second time we get here.
-		id := uint32(len(p.decls))
-		typ := d.Spec.(*ast.TypeSpec).Type
-		decl = &schema.Decl{
-			Id:   id,
-			Name: d.Name,
-			Doc:  d.Doc,
-			Loc:  parseLoc(d.File, typ),
-			// Type is set below
-		}
-		p.declMap[key] = decl
-		p.decls = append(p.decls, decl)
-		decl.Type = p.resolveType(pkg, d.File, d.Spec.(*ast.TypeSpec).Type)
-	}
-
-	return &schema.Type{Typ: &schema.Type_Named{
-		Named: &schema.Named{Id: decl.Id},
-	}}
-}
-
```

Comment:
This was wholesale moved into the new `schema_go1.16.go` file, which a modified version sitting in `schema_go1.18.go`